### DISCUSSION TITLE
CoAP didnt get ACK after receive resource

### DISCUSSION THREAD

**Discussion by AnotherDechathon** [openthread/openthread#10305] [Q&A] [answered]
Hello, I just new to OpenThread and CoAP and got some problem here

I create Thread network by using
 - 1 OTBR using Rpi with NRF52840 Dongle and using OTBR as client
 - 2 MTD using NRF52840 Dongle as server

I was able to form network from those 3 devices and able to use CoAP and be able to see the message "coap reponse from <MTD ipaddr> with payload <payload>" on client side (OTBR) and see message "coap response sent" on server side (MTD), but when I use Tshark to see the network traffic while using CoAP I notice that it only show CON but there wasnt ACK afterward

am not so sure if this issue is similar to https://github.com/openthread/openthread/issues/2733 or https://github.com/openthread/openthread/issues/3143

Thank in advance.

**Comment by jwhui** [comment]
Is the response piggybacked on the ack? CoAP does not require the response and ack to be separate messages.

**Reply by AnotherDechathon** [reply]
Hi, thank for the reply @jwhui 

First of all, I wanna say sorry, I was misunderstood that the ACK wasnt respond back to server, also I might misunderstand some concept of CoAP and more, please correct me if am wrong

after some testing and using sniffer I was able to see what is really going on, the CoAP was piggybacked and ACK was able to send back to server but for some reason it cant be seen on OTBR/server via interface wpan0, it seem like I wasnt able to see all incoming traffic on OTBR/server, is there anyway to fix that or was it should be like that
(P.S. I want to monitor network traffic on OTBR and use that data to process the usage of network on other Thread device (FTD/MTD) without using sniffer)

and one more question, from wireshark picture No. 1-4 and 11-14, why after NON it get ACK

Order of command that was use

```
(On server side)
coap get fe80::f07a:6f7b:2b65:e49b test
coap get fe80::f07a:6f7b:2b65:e49b test con

(On client side)
coap get fe80::88bd:e3eb:5d19:ff71 test2
coap get fe80::88bd:e3eb:5d19:ff71 test2 con
```

**Sniffer pcap** (sniff from outer Thread network device that was mention from OG question)
[sniffer.zip](https://github.com/openthread/openthread/files/15485750/sniffer.zip)
Network key for IEEE decryption : 32348212473af7ae4ea30cb93a918dee
![image](https://github.com/openthread/openthread/assets/95971334/e2b683c9-6be8-4bc6-bc31-c356184be064)


**Current output** (tshark on OTBR/server via interface wpan0)
![image](https://github.com/openthread/openthread/assets/95971334/40e01441-66c7-4497-a173-3333f86328ef)


**Expected output**
be able to get  traffic of CON, NON and ACK via interface wpan0 on OTBR/server like from sniffer that was use on wireshark

Sorry for my bad English
Thank in advance.

**Reply by jwhui** [reply, marked as answer]
OT CoAP CLI uses OpenThread's CoAP client. OpenThread is handling the CoAP messages before they even reach the Linux network stack. As a result, you won't be able to observe them on the Linux network interface.

**Reply by AnotherDechathon** [reply]
Oh I understand now, thank you very much
